tests/unit/api/test_metadata.py

```diff
@@ -0,0 +1,189 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+import json
+
+import pretend
+from fastapi import status
+
+
+class TestPostMetadata:
+    def test_post_metadata(self, test_client, monkeypatch, token_headers):
+        url = "/api/v1/metadata/"
+
+        mocked_check_metadata = pretend.call_recorder(lambda: True)
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.is_bootstrap_done",
+            mocked_check_metadata,
+        )
+
+        mocked_async_result = pretend.stub(state="SUCCESS")
+        mocked_repository_metadata = pretend.stub(
+            apply_async=pretend.call_recorder(lambda *a, **kw: None),
+            AsyncResult=pretend.call_recorder(lambda *a: mocked_async_result),
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.repository_metadata",
+            mocked_repository_metadata,
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.get_task_id", lambda: "123"
+        )
+
+        with open(
+            "tests/data_examples/metadata/update-root-payload.json"
+        ) as f:
+            f_data = f.read()
+
+        payload = json.loads(f_data)
+        response = test_client.post(url, json=payload, headers=token_headers)
+
+        assert response.status_code == status.HTTP_202_ACCEPTED
+        assert response.url == f"{test_client.base_url}{url}"
+        assert response.json() == {
+            "message": "Metadata rotation accepted.",
+            "data": {"task_id": "123"},
+        }
+
+    def test_post_metadata_without_bootstrap(
+        self, test_client, monkeypatch, token_headers
+    ):
+        url = "/api/v1/metadata/"
+
+        mocked_check_metadata = pretend.call_recorder(lambda: False)
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.is_bootstrap_done",
+            mocked_check_metadata,
+        )
+
+        with open(
+            "tests/data_examples/metadata/update-root-payload.json"
+        ) as f:
+            f_data = f.read()
+
+        payload = json.loads(f_data)
+        response = test_client.post(url, json=payload, headers=token_headers)
+
+        assert response.status_code == status.HTTP_200_OK
```

Comment:
Looking into the status code I am wondering:
shouldn't we return non 200 status code when bootstrap doesn't exist?
I am asking because the `POST` operation cannot be completed successfully.
The same question is for the REST API calls where we return status code 200 when bootstrap is `False` and it should be `True`.